src/apps/beefy/safety.ts

```diff
@@ -0,0 +1,77 @@
+import { TFunction } from 'i18next'
+import { Safety } from '../../types/positions'
+import { MAX_SCORE, RISKS, CATEGORIES } from './safetyConfig'
+import { logger } from '../../log'
+import { BaseBeefyVault } from './api'
+
+// From https://github.com/beefyfinance/beefy-v2/blob/3690e105c4bb98afcf06f9c3e385d13cc23af5cd/src/helpers/safetyScore.tsx
+const calcRisk = (risks: string[]) => {
+  const categories: Record<string, string[]> = {}
+  for (const c of Object.keys(CATEGORIES)) {
+    categories[c] = []
+  }
+
+  // reverse lookup
+  risks.forEach((risk) => {
+    // should never happen with check below, but leaving as is from beefy codebase
+    if (!(risk in RISKS)) {
+      return
+    }
+
+    // should never happen with type safety, but leaving as is from beefy codebase
+    const cat = RISKS[risk].category
+    if (!(cat in CATEGORIES)) {
+      return
+    }
+
+    categories[cat].push(risk)
+  })
+
+  // reduce & clamp
```

Comment:
What does clamp mean?

Reply:
it's the math.min if the score goes above 1, taken literally from beefy code